cou/apps/base.py

```diff
@@ -519,57 +528,76 @@ def _get_upgrade_current_release_packages_step(
         return step
 
     def _get_refresh_charm_step(self, target: OpenStackRelease) -> PreUpgradeStep:
-        """Get step for refreshing the current channel.
+        """Get step for refreshing the charm.
 
-        This function also identifies if charm comes from charmstore and in that case,
-        makes the migration.
-        :param target: OpenStack release as target to upgrade.
+        :param target: OpenStack release as target to upgrade
         :type target: OpenStackRelease
         :raises ApplicationError: When application has unexpected channel.
-        :return: Step for refreshing the charm.
+        :return: Step for refreshing the charm
         :rtype: PreUpgradeStep
         """
-        if not self.can_upgrade_to:
-            return PreUpgradeStep()
+        if self.is_from_charm_store:
+            return self._get_charmhub_migration_step()
+        if self.channel == LATEST_STABLE:
+            return self._get_change_to_openstack_channels_step()
+        if self._need_to_refresh(target):
```

Comment:
nit: we are using the word `refresh` as a broader meaning in this function, so it's unclear what `_need_to_refresh` function would do from the name. Could you rename it to something like `_need_in_channel_refresh`?

Reply:
I've changed to `_need_current_channel_refresh`. I think that makes it clear. Thanks

Reply:
Thanks